scripts/generate-icons.js

```diff
@@ -0,0 +1,87 @@
+const path = require('node:path');
+const { readdir, writeFile } = require('node:fs/promises');
+const { readFile } = require('node:fs/promises');
+const { pascalCase, pascalCaseTransformMerge } = require('pascal-case');
+
+const PREFIX = 'ri-';
+const TARGET = 'src/consts/icons.ts';
+
+function resolveRoot(...dir) {
+  return path.resolve(__dirname, '..', ...dir);
+}
+
+function resolveRemixIconDir(...dir) {
+  return path.resolve(
+    __dirname,
+    '..',
+    'node_modules',
+    'remixicon',
+    'icons',
+    ...dir
+  );
+}
+
+async function loop(data, cb) {
+  for (const el of data) {
+    await cb(el);
+  }
+}
+
+const pattern =
+  /<svg viewBox="0 0 24 24" xmlns="http:\/\/www\.w3\.org\/2000\/svg"><path d="([^"]+)"\/><\/svg>/;
+
+function getPathFromSvgString(svg) {
+  return svg.replace(pattern, `$1`);
+}
+
+async function collectAllIconMetas() {
+  const res = [];
+  const categoryDirs = await readdir(resolveRemixIconDir());
+
+  await loop(categoryDirs, async (categoryDir) => {
+    const iconsDirs = await readdir(resolveRemixIconDir(categoryDir));
+
+    await loop(iconsDirs, async (iconName) => {
+      const name = PREFIX + path.basename(iconName).replace('.svg', '');
+      const generatedName = pascalCase(name, {
+        transform: pascalCaseTransformMerge,
+      });
+      const svg = await readFile(
+        resolveRemixIconDir(categoryDir, iconName),
+        'utf8'
+      );
+      const svgPath = getPathFromSvgString(svg);
+
+      res.push({
+        name,
+        generatedName,
+        svgPath,
+      });
+    });
+  });
+
+  return res;
+}
+
+async function generate() {
+  const metadata = await collectAllIconMetas();
+  const fileContent = await collectIconConstantString(metadata);
+  await writeFile(resolveRoot(TARGET), fileContent, 'utf8');
+}
+
+async function collectIconConstantString(icons) {
+  let iconsFile = '// Generated by scripts/generate-icons.js\n';
+  iconsFile += `/* eslint-disable max-lines */\n`;
```

Comment:
We should do this: 
/* eslint-disable */
/* prettier-ignore */

Also maybe we can get some ideas about this from https://github.com/unjs/unimport or https://github.com/antfu/unplugin-auto-import. Maybe we can use some of their logic/ideas for the generation